client/src/document/toolbar/flaws.tsx

```diff
@@ -38,21 +36,22 @@ export function ToggleDocumentFlaws({ flaws }: Pick<Doc, "flaws">) {
     }
   }, [location, navigate, show]);
 
-  const flatFlaws: FlatFlaw[] = Object.entries(flaws)
+  const flawsCounts: FlawCount[] = Object.entries(doc.flaws)
```

Comment:
nit: I'd forego explicit typing if the correct type can be infered. Try removing the explicit typing and then hover over the variable in your editor (or on [this playground](https://www.staging-typescript.org/play?ts=4.0.0-dev.20200625#code/MYewdgzgLgBAZgGwIYHcDCICuYoRgXhgHkAjAKwFNgoA6CnAJwEsKIAKAbxOABMAuGAG0AjABoATKIDMAXVEwETMAGsIAkTIC+ASgBQMAzBoBbJAAc2bQWCTGK8pNUxIEAMWQoIM7QQB8MTn1DYJs7USDgg1BsKAFHKGc3DwgaBHoAcygACwiYHT1gmggQBihLJHkSH3x-EhponBgAWhgkeqwcbQBuIA)) to check whether the definitions match.

Reply:
Or I guess in this case you'll get TS to confirm that the shape matches later anyway, when you actually pass it into <Flaws>